client/sftp_client.py

```diff
@@ -196,202 +614,749 @@ def load_dirs(self, path, parent=None):
+		return os.path.isdir(path)
+
+	def _already_exists_all(self, path):
+		return os.path.exists(path)
 
-	def _get_is_folder(self, fullname):
-		return stat.S_ISDIR(os.stat(fullname).st_mode)
+	def _rename_file(self, _iter, path):
+		os.rename(self._tv_model[_iter][2], path)  # pylint: disable=unsubscriptable-object
 
-	def _get_file_size(self, fullname):
-		return os.path.getsize(fullname)
+	@handle_permission_denied
+	def _make_file(self, path):
+		os.makedirs(path)
 
+	def _get_raw_time(self, fullname):
+		return os.path.getmtime(fullname)
+
+	@handle_permission_denied
 	def delete(self, model, treeiter):
-		try:
-			if model[treeiter][5] == -1:
-				shutil.rmtree(model[treeiter][2])
-			else:
-				os.remove(model[treeiter][2])
-		except OSError:
-			gui_utilities.show_dialog_error('Permission Denied', self.application.get_active_window())
+		if model[treeiter][5] == -1:
+			shutil.rmtree(model[treeiter][2])
 		else:
-			gui_utilities.show_dialog_warning('Successfully deleted ' + model[treeiter][2], self.application.get_active_window())
-			self.refresh()
+			os.remove(model[treeiter][2])
+		gui_utilities.show_dialog_warning('Successfully deleted ' + model[treeiter][2], self.application.get_active_window())
+		self.refresh()
+		if self.parentless:
+			self._tv_model.clear()
+			self.load_dirs(self.default_directory)
+
+	@handle_permission_denied
+	def remove_by_folder_name(self, name):
+		shutil.rmtree(name)
+
+	@handle_permission_denied
+	def remove_by_file_name(self, name):
+		os.remove(name)
+
+	def walk(self, src_file, src, commands, local_name, old_files):
+		for walker in os.walk(src_file):
+			walker = list(walker)
+			temp = walker[0].split('/')
+			loc = temp.index(local_name)
+			# Parse the name relative to the directory selected, i.e /home/osboxes/Music -> /osboxes/Music if osboxes was selected
+			parsed_name = '/'.join(temp[loc:])
+			old_files[parsed_name] = walker[0]
+			walker[0] = parsed_name
+			commands.append(walker)
 
 class RemoteDirectory(DirectoryBase):
+	"""
+	Remote Directory object that defines private methods for rendering remote data
+	using Paramiko's SFTP functionality.
+	"""
+	transfer_direction = 'download'
 	treeview_name = 'treeview_remote'
-	def __init__(self, builder, logger, download_button, application, ftp):
-		_treeview = builder.get_object('SFTPClientGUI.' + self.treeview_name)
-		super(RemoteDirectory, self).__init__(_treeview, logger)
+	def __init__(self, builder, application, ftp, ssh):
 		self.ftp = ftp
-		self.download_button = download_button
-		self.application = application
-		self.def_dir = self.application.config['server_config']['server.web_root']
-		self.load_dirs(self.def_dir)
-		self.treeview_local.set_model(self._tvmodel)
+		self.ssh = ssh
+		self.stat = ftp.stat
+		super(RemoteDirectory, self).__init__(builder, application, application.config['server_config']['server.web_root'])
 
+	### FIX ME: Paramiko has no built in command for checking file permissions
```

Comment:
If the permissions can't be properly checked then they should be displayed in their entirety. It sounds like the entire plugin should use the model of `rwxrwxrwx` to display the permissions for user, group and everyone.